Userland/Libraries/LibWeb/SVG/SVGPathElement.cpp

```diff
@@ -390,8 +390,33 @@ float PathDataParser::parse_fractional_constant()
 float PathDataParser::parse_number()
 {
     auto number = parse_fractional_constant();
-    if (match('e') || match('E'))
-        TODO();
+
+    if (!match('e') && !match('E'))
+        return number;
+    consume();
+
+    auto exponent_sign = parse_sign();
+
+    StringBuilder exponent_builder;
+    while (!done() && isdigit(ch()))
+        exponent_builder.append(consume());
+    VERIFY(exponent_builder.length() > 0);
+
+    auto exponent = exponent_builder.to_string().to_int().value();
+
+    if (number == 0) // fast path: if the number is 0 theres no point to computing the exponentiation
+        return number;
+
+    if (exponent_sign < 0) {
+        for (int i = 0; i < exponent; ++i) {
+            number /= 10;
+        }
+    } else if (exponent_sign > 0) {
+        for (int i = 0; i < exponent; ++i) {
+            number *= 10;
```

Comment:
This can overflow, what should happen in that case?

Reply:
these are floating point values, so "overflow" just means being +INFINITY, which would likely result in some very bad rendering, but im not sure if thats something the parser should handle